crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -2591,3 +2617,28 @@ pub(super) fn hint_if_stdlib_submodule_exists_on_other_versions(
 
     add_inferred_python_version_hint_to_diagnostic(db, &mut diagnostic, "resolving modules");
 }
+
+/// Suggest a name from `existing_names` that is similar to `wrong_name`.
+pub(super) fn did_you_mean<S: AsRef<str>, T: AsRef<str>>(
```

Comment:
I copied this function more or less 1:1 from another project of mine. It's not like I have a huge amount of experience with the heuristic here, but I did some iterations on it and it usually gives decent results.